src/toolkits/object_detection/object_detector.cpp

```diff
@@ -353,23 +353,22 @@ void object_detector::load_version(iarchive& iarc, size_t version) {
 void object_detector::import_from_custom_model(variant_map_type model_data,
                                                size_t version) {
   auto it = model_data.find("_model");
-  flex_dict model = variant_get_value<flex_dict>(it->second);
-  model_data.erase(it);
+  const flex_dict& model = variant_get_value<flex_dict>(it->second);
   auto it2 = model_data.find("_grid_shape");
+  if (it2 == model_data.end()) {
+    log_and_throw("The provided model must contain field '_grid_shape'!");
```

Comment:
I don't think this should be an exception. If the grid shape is missing, for now we can assume the 13-by-13 grid from the original implementation

(It should, however, be an exception if "_model" is not found)